lib/config.js

```diff
@@ -229,6 +230,16 @@ function getLocalConfig(thisConfig, directory) {
             continue;
         }
 
+        // Check for root flag
+        if (localConfig.root === true) {
+            rootPath = path.dirname(localConfigFile);
+        }
+
+        // If root flag is set, don't consider file if it is above root
+        if (rootPath && localConfigFile.indexOf(rootPath) === -1) {
```

Comment:
Thanks for the tip, that's good to know.  The main issue with indexOf seems to be that on Windows you can have the same path with different capitalizations.  However, in this case where the paths being compared are all returned from the same process, is that really a concern?  I'd hate to introduce a dependency if one isn't needed.